### PR TITLE
add sql for migrating access log

### PR DESCRIPTION
**Pull request by wy65701436**
1, loop each access log, change to resource/resource_type, and insert into audit log
2, loop each first push operation, change it to create repository and insert into audit log.

Signed-off-by: wang yan <wangyan@vmware.com>

### CODE REVIEW

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql [hunk 1]
```diff
@@ -184,3 +184,43 @@ CREATE TABLE audit_log
...
+/*migrate access log to audit log*/
+/*TODO drop table access_log?*/
+DO $$
+DECLARE
+    access RECORD;
+BEGIN
+    FOR access IN SELECT * FROM access_log
+    LOOP
+      /*insert project create and delete*/
+      IF access.operation = 'create' OR access.operation = 'delete' THEN
```
Comment (ywk253100) [inline make/migrations/postgresql/0030_2.0.0_schema.up.sql]: The resources deleted can be project and image

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql [hunk 2]
```diff
@@ -184,3 +184,43 @@ CREATE TABLE audit_log
...
+          INSERT INTO audit_log (project_id, operation, resource_type, resource, username, op_time) VALUES (access.project_id, access.operation, 'project', access.repo_name, access.username, access.op_time);
+      END IF;
```
Comment (ywk253100) [inline make/migrations/postgresql/0030_2.0.0_schema.up.sql]: Using `if xxx else if xxx end` can reduce the comparing times

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql [hunk 3]
```diff
@@ -184,3 +184,43 @@ CREATE TABLE audit_log
...
+      /*insert push image -> create artifact*/
+      IF access.operation = 'push' THEN
+          INSERT INTO audit_log (project_id, operation, resource_type, resource, username, op_time) VALUES (access.project_id, 'create', 'artifact', CONCAT(access.repo_name,'/',access.repo_tag), access.username, access.op_time);
```
Comment (ywk253100) [inline make/migrations/postgresql/0030_2.0.0_schema.up.sql]: Let's use `push` as the operation as the commands are `push` and the operation times are also `push time`?

Comment (ywk253100) [inline make/migrations/postgresql/0030_2.0.0_schema.up.sql]: Should be `repo_name:repo_tag`

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql [hunk 4]
```diff
@@ -184,3 +184,43 @@ CREATE TABLE audit_log
...
+      END IF;
+      /*insert pull image -> pull tag*/
+      IF access.operation = 'pull' THEN
+          INSERT INTO audit_log (project_id, operation, resource_type, resource, username, op_time) VALUES (access.project_id, 'pull', 'tag', CONCAT(access.repo_name,'/',access.repo_tag), access.username, access.op_time);
```
Comment (ywk253100) [inline make/migrations/postgresql/0030_2.0.0_schema.up.sql]: The resource type should be `artifact`

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql [hunk 5]
```diff
@@ -184,3 +184,43 @@ CREATE TABLE audit_log
...
+      END IF;
+    END LOOP;
+END $$;
+
+/*migrate access log to audit log*/
+/*user the first push operation to generate the create repository log*/
+DO $$
```
Comment (ywk253100) [inline make/migrations/postgresql/0030_2.0.0_schema.up.sql]: If we generate the audit log for repository creation, how about the repository deletion? 
We cannot handle the case that the image is pushed, then deleted, then pushed again, then deleted, which means the repository is created, then deleted and created, deleted

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql (hunk 3 above)
Comment (wy65701436) [reply make/migrations/postgresql/0030_2.0.0_schema.up.sql]: updated.

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql (hunk 5 above)
Comment (wy65701436) [reply make/migrations/postgresql/0030_2.0.0_schema.up.sql]: that's why it just handles the first creation

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql (hunk 4 above)
Comment (wy65701436) [reply make/migrations/postgresql/0030_2.0.0_schema.up.sql]: the resource type is `artifact` when pull by digest? If pull by tag, the type shouldn't be `tag`?

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql (hunk 3 above)
Comment (wy65701436) [reply make/migrations/postgresql/0030_2.0.0_schema.up.sql]: in the new OCI framework, do we have `push` action? Isn't it create an artifact?

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql (hunk 2 above)
Comment (wy65701436) [reply make/migrations/postgresql/0030_2.0.0_schema.up.sql]: OK

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql (hunk 1 above)
Comment (wy65701436) [reply make/migrations/postgresql/0030_2.0.0_schema.up.sql]: sure

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql (hunk 4 above)
Comment (wy65701436) [reply make/migrations/postgresql/0030_2.0.0_schema.up.sql]: updated.

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql (hunk 5 above)
Comment (wy65701436) [reply make/migrations/postgresql/0030_2.0.0_schema.up.sql]: removed.

**Pull request merged**